util/util.go

```diff
@@ -16,6 +17,65 @@ func init() {
 	ByteOrder = binary.LittleEndian
 }
 
+// BackoffPrint allows to print a statement but not too often.
+type BackoffPrint struct {
+	nextPrint    time.Time
+	frequency    time.Duration
+	minFrequency time.Duration
+	maxFrequency time.Duration
+	factor       int
+}
+
+// NewBackoffPrint creates an instance of BackoffPrint.
+// The `minFrequency` indicates how frequently BackoffPrint.Ok() can return true.
+// When Ok() returns true, the allowed frequency is multiplied by `factor`. The
+// resulting frequency is capped by `maxFrequency`.
+func NewBackoffPrint(minFrequency time.Duration, factor int, maxFrequency time.Duration) *BackoffPrint {
+	return &BackoffPrint{
+		frequency:    minFrequency,
+		minFrequency: minFrequency,
+		maxFrequency: maxFrequency,
+		factor:       factor,
+	}
+}
+
+// Ok returns true for the first time it is invoked after creation of the object
+// or call to Reset(), or after an amount of time (based on the last success
+// and the allowed frequency) has elapsed.
+// When at the maximum frequency, if this call is made after a delay at least
+// equal to 3x the max frequency (or in other words, 2x after what was the target
+// for the next print), then the object is auto-reset.
+func (bp *BackoffPrint) Ok() bool {
+	if bp.nextPrint.IsZero() {
+		bp.nextPrint = time.Now().Add(bp.minFrequency)
+		return true
+	}
+	if time.Now().Before(bp.nextPrint) {
```

Comment:
Actually no, since there are returns. Each time time.Now() is invoked, it is once per code path.
Unless I change `time.Since()` to do something like `now.Sub(bp.nextPrint)`, which I could.